### PR TITLE
Add observability content to DevEx

### PR DESCRIPTION
**Pull request by cteicher-m**

### CODE REVIEW

File: developer-experience/readme.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ New team members are uniquely positioned to identify instances of undocumented [
 
 The following outline examples of several strategies that can be adopted to promote a positive developer experience. It is expected that each team should define what a positive dev experience means within the context of their project. Additionally, refine that over time via feedback mechanisms such as sprint and project retrospectives.
 
+### Observability
```
Comment (fnocera) [inline developer-experience/readme.md:88]: I would perhaps move observability further down the list vs having it as the first in the strategies to have a positive DevEx.

File: developer-experience/readme.md [hunk 2]
```diff
@@ -85,6 +85,12 @@ New team members are uniquely positioned to identify instances of undocumented [
 
 The following outline examples of several strategies that can be adopted to promote a positive developer experience. It is expected that each team should define what a positive dev experience means within the context of their project. Additionally, refine that over time via feedback mechanisms such as sprint and project retrospectives.
 
+### Observability
+
+[Observability](../observability/readme.md) alleviates unforeseen challenges for the developer in a complex distributed system. It identifies project bottlenecks quicker and with more precision, enhancing performance as the developer seeks to deploy code changes. Adding observability improves the experience when identifying and resolving bugs or broken code. This positive DevEx results in fewer or less severe current and future production failures.
```
Comment (fnocera) [inline developer-experience/readme.md:90]: Would edit the last sentence a little to remove the words "positive DevEx" as it sounds a bit funny eg: "This also results in fewer or less severe....".

**Review by fnocera — APPROVED** [review]

File: developer-experience/readme.md (hunk 2 above)
Comment (charleszipp) [reply developer-experience/readme.md:90]: +1

File: developer-experience/readme.md [hunk 3]
```diff
@@ -85,6 +85,12 @@ New team members are uniquely positioned to identify instances of undocumented [
...
 
+### Observability
+
+[Observability](../observability/readme.md) alleviates unforeseen challenges for the developer in a complex distributed system. It identifies project bottlenecks quicker and with more precision, enhancing performance as the developer seeks to deploy code changes. Adding observability improves the experience when identifying and resolving bugs or broken code. This positive DevEx results in fewer or less severe current and future production failures.
+
+There are many observability strategies a developer can use alongside best engineering practices. These resources improve the DevEx by ensuring a shared view of the complex system throughout the entire lifecycle. Observaibility in code via logging, exception handling and exposing of relevant application metrics for example, promotes the consistent visibility of real time performance. The observability pillars, [logging](../observability/pillars/logging.md), [metrics](../observability/pillars/metrics.md), and [tracing](../observability/pillars/tracing.md), detail when to enable each of the three specific types of observability.
```
Comment (charleszipp) [inline developer-experience/readme.md:92]: Good use of the links! Will help avoid duplicating content that could easily become stale.

**Review by charleszipp — APPROVED** [review]

**Pull request merged**